Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteScrollingTreeMac.mm

```diff
@@ -85,6 +85,20 @@
     });
 }
 
+void RemoteScrollingTreeMac::scrollingTreeNodeDidBeginScrollSnapping(ScrollingNodeID nodeID)
+{
+    RunLoop::main().dispatch([this, nodeID] {
+        RemoteScrollingTree::scrollingTreeNodeDidBeginScrollSnapping(nodeID);
+    });
+}
+
+void RemoteScrollingTreeMac::scrollingTreeNodeDidEndScrollSnapping(ScrollingNodeID nodeID)
+{
+    RunLoop::main().dispatch([this, nodeID] {
```

Comment:
Not new to this patch but capturing `this` doesn't seem safe since RemoteScrollingTreeMac isn't a singleton object and gets destroyed when the WebPageProxy gets destroyed.

Reply:
Good catch — should probably be a `Ref`? I can fix it here (and in the other call sites) in this patch.